lib/matplotlib/axes/_base.py

```diff
@@ -1687,6 +1697,15 @@ def axis(self, *v, **kwargs):
             ymin, ymax = self.get_ylim()
             return xmin, xmax, ymin, ymax
 
+        if isinstance(v, bool):
```

Comment:
Since `v` is defined to collect all positional argument (`def axis(self, *v, ...)`), it is a tuple. You have to 

~~~
if len(v) == 1 and isinstance(v[0], bool):
~~~

similar to the `str` case above.